### PR TITLE
Custom auth polish

### PR DESCRIPTION
**Pull request by bretambrose**
* Encode token signature if needed
* Update documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: source/iot/Mqtt5Client.cpp [hunk 1]
```diff
@@ -729,7 +730,17 @@ namespace Aws
 
         Mqtt5CustomAuthConfig &Aws::Iot::Mqtt5CustomAuthConfig::WithTokenSignature(Crt::String tokenSignature)
         {
-            m_tokenSignature = std::move(tokenSignature);
+            if (tokenSignature.find('%') != tokenSignature.npos)
+            {
+                // We can assume that a base 64 value that contains a '%' character has already been uri encoded
+                m_tokenSignature = tokenSignature;
```
Comment (sfod) [inline source/iot/Mqtt5Client.cpp:736]: nit: `std::move(tokenSignature)`

File: source/iot/Mqtt5Client.cpp [hunk 2]
```diff
@@ -729,7 +730,17 @@ namespace Aws
...
+            }
+            else
+            {
+                m_tokenSignature =
+                    Aws::Crt::Io::EncodeQueryParameterValue(aws_byte_cursor_from_c_str(tokenSignature.c_str()));
```
Comment (sfod) [inline source/iot/Mqtt5Client.cpp:741]: Debatable nit: if change EncodeQueryParameterValue to accept `const String&`, the calling code will be less verbose

File: source/iot/MqttClient.cpp [hunk 3]
```diff
@@ -381,8 +382,21 @@ namespace Aws
...
+                }
+                else
+                {
+                    encodedSignature = Aws::Crt::Io::EncodeQueryParameterValue(
+                        aws_byte_cursor_from_c_str(authorizerSignature.c_str()));
+                }
+
                 usernameString =
-                    AddToUsernameParameter(usernameString, authorizerSignature, "x-amz-customauthorizer-signature=");
+                    AddToUsernameParameter(usernameString, encodedSignature, "x-amz-customauthorizer-signature=");
```
Comment (sfod) [inline source/iot/MqttClient.cpp:399]: nit: `std::move(usernameString)` and `std::move(encodedSignature)`

**Review by sfod — APPROVED** [review]

File: source/iot/MqttClient.cpp [hunk 4]
```diff
@@ -381,8 +382,21 @@ namespace Aws
...
             }
             if (!tokenKeyName.empty() || !tokenValue.empty())
```
Comment (sfod) [inline source/iot/MqttClient.cpp:401]: Similar to aws-crt-nodejs, we can check for a case when tokenKeyName and tokenValue are set while authorizerSignature is empty.

File: source/iot/Mqtt5Client.cpp (hunk 2 above)
Comment (bretambrose) [reply source/iot/Mqtt5Client.cpp:741]: From a utility perspective, taking a byte cursor is quite a bit more useful.

**Pull request merged**